### PR TITLE
resume a change address filing (+ more Resume AR work)

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* /bcgov/entity#982

*Description of changes:*

This is the remainder of the work for the subject ticket. (The first part was PR #194 and was previously merged to allow others to work with the updated code.)

- deleted obsolete getters
- updated POST and PUT
- updated unit tests
- added http status codes package
- updated API URL in config
- hid vs disabled Save and Save & Resume buttons

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: coops-ui/src/views/StandaloneDirectorsFiling.vue [hunk 1]
```diff
@@ -276,9 +276,6 @@ h2
   font-size: 1.125rem;
   font-weight: 500;
 
-#AR-step-1-container, #AR-step-2-container, #AR-step-3-container
```
Comment (katiemcgoff) [inline coops-ui/src/views/StandaloneDirectorsFiling.vue]: leave in

File: coops-ui/src/views/AnnualReport.vue [hunk 2]
```diff
@@ -540,16 +537,26 @@ export default {
...
+        this.toggleFiling('remove', 'OTCDR')
+      }
+    },
+
     directorFormValid (val) {
       this.setValidateFlag()
     },
 
     validated (val) {
-      console.log('AnnualReport, validated =', val)
```
Comment (katiemcgoff) [inline coops-ui/src/views/AnnualReport.vue]: remove

File: coops-ui/src/views/AnnualReport.vue [hunk 3]
```diff
@@ -363,23 +358,32 @@ export default {
...
       this.saveErrorDialog = false
       let changeOfDirectors = null
       let changeOfAddress = null
 
       const header = {
         header: {
           name: 'annualReport',
-          date: this.currentDate
+          date: this.currentDate,
+          submitter: sessionStorage.getItem('USERNAME')
```
Comment (katiemcgoff) [inline coops-ui/src/views/AnnualReport.vue]: take out

File: coops-ui/src/views/AnnualReport.vue [hunk 4]
```diff
@@ -130,16 +128,12 @@
...
-                  @modified="officeModifiedEventHandler($event)"
-                  @valid="officeValidEventHandler($event)"
-                ></RegisteredOfficeAddress>
-              </v-card>
+              <RegisteredOfficeAddress
+                :changeButtonDisabled="!agmDateValid"
+                :legalEntityNumber="corpNum"
+                :addresses.sync="addresses"
+                @modified="officeModifiedEventHandler($event)"
+                @valid="officeValidEventHandler($event)" />
```
Comment (WalterMoar) [inline coops-ui/src/views/AnnualReport.vue]: I think I wrote this originally, yuck.

Just a comment: Next time we're in here, I think we don't need the `($event)` on the callbacks. I think Vue knows to call the named method with the obvious parameter.

File: coops-ui/src/views/AnnualReport.vue [hunk 5]
```diff
@@ -363,23 +358,32 @@ export default {
 
     directorsChange (val) {
       // when directors change, update filing data
-      console.log('AnnualReport, directorsChange =', val)
+      // console.log('AnnualReport, directorsChange =', val)
```
Comment (WalterMoar) [inline coops-ui/src/views/AnnualReport.vue]: In the future can we just delete these? At some point the linter should be set up to complain about commented out code.

File: coops-ui/src/views/AnnualReport.vue [hunk 6]
```diff
@@ -540,16 +537,26 @@ export default {
...
+      }
+    },
+
     directorFormValid (val) {
       this.setValidateFlag()
     },
 
     validated (val) {
-      console.log('AnnualReport, validated =', val)
+      // console.log('AnnualReport, validated =', val)
```
Comment (WalterMoar) [inline coops-ui/src/views/AnnualReport.vue]: Can we delete the watchers that are now doing nothing? Not sure they ever should have been committed.

File: coops-ui/src/views/AnnualReport.vue (hunk 4 above)
Comment (severinbeauvais) [reply coops-ui/src/views/AnnualReport.vue]: Agreed. And yup, I didn't change it ;)

File: coops-ui/src/views/AnnualReport.vue (hunk 5 above)
Comment (severinbeauvais) [reply coops-ui/src/views/AnnualReport.vue]: Yes, I am deleting this (per Katie's review).

File: coops-ui/src/views/AnnualReport.vue (hunk 3 above)
Comment (severinbeauvais) [reply coops-ui/src/views/AnnualReport.vue]: done

File: coops-ui/src/views/AnnualReport.vue (hunk 2 above)
Comment (severinbeauvais) [reply coops-ui/src/views/AnnualReport.vue]: done

File: coops-ui/src/views/StandaloneDirectorsFiling.vue (hunk 1 above)
Comment (severinbeauvais) [reply coops-ui/src/views/StandaloneDirectorsFiling.vue]: done

File: coops-ui/src/views/AnnualReport.vue (hunk 6 above)
Comment (severinbeauvais) [reply coops-ui/src/views/AnnualReport.vue]: Yep, done. Refresh the page to see the new commits.

**Comment by severinbeauvais** [comment]
> It makes me so happy to see ids and refs being removed! Some good cleanup and simplification.

Yes, as long as they're not used. Sometimes (the ids are used by the unit tests.)

**Pull request merged**